File: Easy/Javascript/2441. Largest Positive Integer That Exists With Its Negative.js
/**Given an integer array nums that does not contain any zeros, find the largest positive integer k such that -k also exists in the array.

Return the positive integer k. If there is no such integer, return -1. */

/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxK = function (nums) {
  nums = nums.sort((a, b) => a - b);
  let right = nums.length;
  let left = 0;

  while (left < right) {
    let summation = nums[left] + nums[right];

    if (summation === 0) return nums[right];
    else if (summation < 0) left += 1;
    else right -= 1;
  }

  return -1;
};

console.log(findMaxK([-1,2,-3,3]))

File: Easy/Python/2441. Largest Positive Integer That Exists With Its Negative.py
"""
Given an integer array nums that does not contain any zeros, find the largest positive integer k such that -k also exists in the array.

Return the positive integer k. If there is no such integer, return -1.
"""

from typing import List

class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        nums.sort()
        left = 0
        right = len(nums) - 1

        while left < right:
            summation = nums[left] + nums[right]
            if summation == 0:
                return abs(nums[left])
            elif summation < 0:
                left += 1
            else:
                right -= 1

        return -1